### PR TITLE
feat(webhook): Allow configuring retry handling per status code on webhook monitoring.

### PR DESCRIPTION
**Pull request by Bijnagte**

### CODE REVIEW

File: orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/pipeline/WebhookStage.java [hunk 1]
```diff
@@ -84,7 +85,7 @@ public void onFailureStages(@Nonnull StageExecution stage, @Nonnull StageGraphBu
     public String url;
     public Object payload;
     public Map<String, Object> customHeaders;
-    public List<Integer> failFastStatusCodes;
+    public List<Integer> failFastStatusCodes = List.of(504);
```
Comment (Bijnagte) [inline orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/pipeline/WebhookStage.java:88]: default to having a gateway timeout result in a fail fast by default

Comment (dbyron0) [reply orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/pipeline/WebhookStage.java:88]: Perhaps https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpStatus.html#GATEWAY_TIMEOUT instead to make that more obvious?

File: orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java [hunk 2]
```diff
@@ -119,135 +126,143 @@ public TaskResult execute(StageExecution stage) {
...
   }
 
+  private boolean shouldRetry(HttpStatus statusCode, WebhookStage.StageData stageData) {
+    int status = statusCode.value();
+    var retries = stageData.getRetries();
+
+    if (retries != null && retries.containsKey(status)) {
+      // specific retry limit configured
+      var retryConfig = retries.get(status);
+      long pastStatuses =
```
Comment (jonsie) [inline orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java:250]: nit; `pastStatuses` reads like a list of statuses, maybe change to `attempts`?

**Review by jonsie — APPROVED** [review]
This looks like a nice cleanup and addition to me.

Are you thinking the retry config would be added to the webhook stage UI?

**Comment by Bijnagte** [comment]
Yeah, the UI will need to be updated to allow configuring. That will be my next task.

File: orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java (hunk 2 above)
Comment (Bijnagte) [reply orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java]: I updated this and extracted a method to make it read a little better

**Comment by srekapalli** [comment]
Will we also apply the same retry config for the initial webhook calls or only fo monitor webhook status calls..?

**Comment by Bijnagte** [comment]
I was not sure about the retry on the initial call and if it should be kept distinct since they are different methods, I could certainly reuse the same values though. I also didn't necessarily want to make the full set of changes that would involve that in one PR.

**Pull request merged**